### ISSUE TITLE
In 3DS build, g++ complain about implicit type conversion in mame OPL emulation

### ISSUE THREAD

**Issue by zephray** [sdlpal/sdlpal#95]
Error as follows:
```
c:\devkitpro\msys2\tmp\sdlpal\adplug\mame/ymf262.cpp.h:657:19: error: invalid conversion from 'int*' to 'int32_t*' {aka 'long int*'} [-fpermissive]
   slot->connect = &chip->chanout[slot->conn_enum - 1];
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
c:\devkitpro\msys2\tmp\sdlpal\adplug\mame/ymf262.cpp.h:659:19: error: invalid conversion from 'int*' to 'int32_t*' {aka 'long int*'} [-fpermissive]
   slot->connect = &chip->phase_modulation;
                   ^~~~~~~~~~~~~~~~~~~~~~~
```
Can be fixed by adding -fpermissive to the CXXFLAGS, but I do not think this is the preferred solution here. Adding explicit type conversion would probably hurt other ports (on, e.g. 64 bit systems. Do we have any 16 bit or even 8 bit ports?)

**Comment by louyihua** [comment]
Thanks for reporting this issue.
This is fixed in c8fb6e1 by changing the variables defined as `signed int` into `int32_t`. Since most 32-bit & 64-bit systems treats `int32_t` as an alias of `int`, this change will not break current builds.

**Comment by zephray** [comment]
Looks nice, but there is still few need to be changed for the same reason. I have changed them created a pull request, which should fully address this problem.
